ui/StatusQ/src/StatusQ/Controls/StatusBaseButton.qml

```diff
@@ -84,14 +93,13 @@ Button {
 
     contentItem: RowLayout {
         spacing: root.spacing
-        StatusIcon {
-            Layout.preferredWidth: visible ? root.icon.width : 0
-            Layout.preferredHeight: visible ? root.icon.height : 0
-            icon: root.icon.name
-            rotation: root.asset.rotation
-            opacity: !loading && root.icon.name !== ""
-            visible: root.icon.name !== ""
-            color: root.icon.color
+        Loader {
+            id: iconLoader
+
```

Comment:
you can keep the loader, but please use instead of exposing the whole component, use a boolean to determine whether to load StatusIcon or StatusRoundedIcon. Since both will be end up using root.icon's properties you can use [Binding](https://doc.qt.io/qt-6/qml-qtqml-binding.html) to bind the loaded items corresponding properties when the item is successfully loaded.

Reply:
so far this is the only case where we need other component than the StatusIcon, if in the future we see we need to be more flexible we can update accordingly

Reply:
yeah, the thing is that we do not know at this point if we'll need further changes and I'd like to avoid adding unnecessary stuff in components which won't be used at the end as this creates hard to read and unmaintainable APIs. other than this I see that both components StatusIcon and StatusRoundIcon will use the same bindings in all cases, so the only difference in this known use case is the component loaded, nothing more. @caybro would like your opinion